src/test.rs

```diff
@@ -28,15 +28,15 @@ fn new_max_heap() -> Heap {
 fn empty() {
     let mut heap = Heap::empty();
     let layout = Layout::from_size_align(1, 1).unwrap();
-    assert!(heap.allocate_first_fit(layout.clone()).is_err());
+    assert!(heap.allocate_first_fit(layout.clone()) as *mut u8 == core::ptr::null_mut());
 }
 
 #[test]
 fn oom() {
     let mut heap = new_heap();
     let layout = Layout::from_size_align(heap.size() + 1, align_of::<usize>());
     let addr = heap.allocate_first_fit(layout.unwrap());
-    assert!(addr.is_err());
+    assert!(addr as *mut u8 == core::ptr::null_mut());
```

Comment:
Maybe we can encapsulate the `== core::ptr::null_mut()` check in some `is_err` / `is_ok` functions to avoid the boilerplate? It might also make sense to move all the tests in an inline `tests` submodule.

Reply:
That's a good idea